### PR TITLE
Sign up backend integration

### PR DESCRIPTION
**Pull request by mariadoan247**
User authentication and validation logic such as isEmpty or alreadyExsits is put in place!

### CODE REVIEW

**Review by rebekahre — APPROVED** [review]
gj

File: backend/routes/users.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  const { errors, isValid } = validateRegisterInput(req.body.document);
+
+  // Check validation
+  if (!isValid) {
+    return res.status(400).json(errors);
+  }
+  
+  User.findOne({ email: req.body.document.email }).then(user => {
+    if (user) {
+      return res.status(400).json({ email: "Email already exists" });
```
Comment (n-grounds) [inline backend/routes/users.js:30]: Should it be the `email` field that contains the error message `"Email already exists"` or should it be a field named something like `errorMsg`?

File: backend/routes/users.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    } else {
+      const newUser = new User({
+        name: req.body.document.name,
+        email: req.body.document.email,
+        password: req.body.document.password
+      });
+      
+      // Hash password before saving in database
+      bcrypt.genSalt(10, (err, salt) => {
```
Comment (n-grounds) [inline backend/routes/users.js:39]: the `err` value here is never checked, is that a hole in logic?

Comment (mariadoan247) [reply backend/routes/users.js:39]: I will check this asap and see if it is a hole in the logic.

File: backend/routes/users.js (hunk 1 above)
Comment (mariadoan247) [reply backend/routes/users.js:30]: This was an oversight on my part. I am adding this error handeling into validation/signUp.js instead.

**Pull request merged**